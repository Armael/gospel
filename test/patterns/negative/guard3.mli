val f : int -> int
(*@ r = f x
  requires match x with
           | 1i -> true
           | _ when 1 = 2 -> false
*)
