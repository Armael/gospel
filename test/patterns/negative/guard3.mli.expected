
*******************************
********** Parsed file ********
*******************************
val f : int -> int[@@gospel
                    {| r = f x
  requires match x with
           | 1i -> true
           | _ when 1 = 2 -> false
|}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Stdlib *)

(*@ open Gospelstdlib *)

val f : int -> int
(*@ r = f x
    requires ...
     *)
File "guard3.mli", line 3, characters 11-82:
Error: This pattern-matching may not be exhaustive because of the guard.
       Here is an example of a case that may not be matched:
         0i.
